nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/RecordFilterChain.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * (C) Copyright 2019 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.lib.stream.computation;
+
+/**
+ * A chain of record filters.
+ *
+ * @since 11.1
+ */
+public interface RecordFilterChain extends RecordFilter {
+
+    RecordFilterChain addFilter(RecordFilter filter);
```

Comment:
Maybe specify in Javadoc if the filter is added before or after existing ones (and what it means exactly in terms of processing, given that filters themselves have before- and after-hooks).